core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -108,6 +108,31 @@ public static DataFile fromManifest(ManifestFile manifest) {
         .build();
   }
 
+  public static DataFile fromPositionDelete(DeleteFile deleteFile, PartitionSpec spec) {
+    Preconditions.checkArgument(deleteFile.content().equals(FileContent.POSITION_DELETES),
+        "Expected positional delete file, found %s", deleteFile.content());
+    Metrics metrics = new Metrics(
+        deleteFile.recordCount(),
+        deleteFile.columnSizes(),
+        deleteFile.valueCounts(),
+        deleteFile.nullValueCounts(),
+        deleteFile.nanValueCounts(),
+        deleteFile.lowerBounds(),
+        deleteFile.upperBounds()
+    );
+
+    return DataFiles.builder(spec)
+        .withEncryptionKeyMetadata(deleteFile.keyMetadata())
+        .withFormat(deleteFile.format())
+        .withFileSizeInBytes(deleteFile.fileSizeInBytes())
+        .withPath(deleteFile.path().toString())
+        .withPartition(deleteFile.partition())
+        .withRecordCount(deleteFile.recordCount())
+        .withSortOrder(SortOrder.unsorted())
```

Comment:
Shall we propagate the sort order? We can add `withSortOrderId` to the builder.

Reply:
In my test, sort order id doesnt seem to be populated on delete files (returns null), but added a conditional set in case it is populated.